src/handlers/QuotingHandler.js

```diff
@@ -0,0 +1,192 @@
+const { AuditEventAction } = require('@mojaloop/event-sdk')
+const { Enum } = require('@mojaloop/central-services-shared')
+const { reformatFSPIOPError } = require('@mojaloop/central-services-error-handling').Factory
+
+const { ErrorMessages } = require('../lib/enum')
+const { getSpanTags } = require('../lib/util')
+const dto = require('../lib/dto')
+
+const { FSPIOP } = Enum.Http.Headers
+
+class QuotingHandler {
+  constructor (deps) {
+    this.quotesModelFactory = deps.quotesModelFactory
+    this.bulkQuotesModelFactory = deps.bulkQuotesModelFactory
+    this.logger = deps.logger
+    this.config = deps.config
+    this.cache = deps.cache
+    this.tracer = deps.tracer
+    this.handleMessages = this.handleMessages.bind(this)
+  }
+
+  async handleMessages(error, messages) {
+    // think, if we need to add Metrics.getHistogram here
+    if (error) {
+      this.logger.error(`${ErrorMessages.consumingErrorFromKafka}: ${error.message}`, { error })
+      throw reformatFSPIOPError(error)
+    }
+
+    const results = await Promise.allSettled(
+      messages.map(msg => this.defineHandlerByTopic(msg))
+    )
+    this.logger.info('handleMessages results:', results)
+
+    return true
+  }
+
+  async defineHandlerByTopic(message) {
+    const { topic, requestData } = dto.requestDataFromMessageDto(message)
+    const { QUOTE, BULK_QUOTE } = this.config.kafkaConfig.CONSUMER
+
+    switch (topic) {
+      case QUOTE.POST.topic:
+        return this.handlePostQuotes(requestData)
+      case QUOTE.PUT.topic:
+        return this.handlePutQuotes(requestData)
+      case QUOTE.GET.topic:
+        return this.handleGetQuotes(requestData)
+      case BULK_QUOTE.POST.topic:
+        return this.handlePostBulkQuotes(requestData)
+      case BULK_QUOTE.PUT.topic:
+        return this.handlePutBulkQuotes(requestData)
+      case BULK_QUOTE.GET.topic:
+        return this.handleGetBulkQuotes(requestData)
+
+      default:
+        this.logger.warn(ErrorMessages.unsupportedKafkaTopic, message)
+    }
+  }
+
+  async handlePostQuotes(requestData) {
+    const { requestId, payload, headers } = requestData
+    const model = this.quotesModelFactory(requestId)
+    let span
+
+    try {
+      span = await this.createSpan(requestData)
+      const result = await model.handleQuoteRequest(headers, payload, span, this.cache)
+      this.logger.debug('handlePostQuotes is done:', { result })
+    } catch (err) {
+      this.logger.error(`error in handlePostQuotes: ${err?.message}`, { requestData })
+      const fspiopError = reformatFSPIOPError(err)
+      const fspiopSource = headers[FSPIOP.SOURCE]
+      await model.handleException(fspiopSource, payload.quoteId, fspiopError, headers, span)
+    }
+
+    return true
+  }
+
+  async handlePutQuotes(requestData) {
+    const { id: quoteId, requestId, payload, headers } = requestData
+    const model = this.quotesModelFactory(requestId)
+    const isError = !!payload.errorInformation
```

Comment:
Sorry, but I don't get what benefits having a separate handlePutQuotesError function gives us. We publish PUT suceess and error payload in the same queue, so what's wrong with handling of 2 types of payload in one processing functions. Also, please take into account, that actual processing is done in a separate model methods

Reply:
We need to reduce amount of if conditions for performance, I found more than one if condition in the flow to check if it is an error callback. That's why I proposed having a new function and calling a separate function based on the message reduces the need of further if conditions in the flow.

Reply:
If you remember, exactly for such reason (to avoid unnecessary checks) I proposed to have a separate queue for each payloads (even for _PUT success_ and _PUT error_). But we agreed to publish PUT payloads to one queue. In this case I don't see a good way to avoid 2 checks (first, check queue, then check a payload).

I think, we can avoid any condition check in our flow, if we'll specify for each topic a particular function (e.g., handlePostQuotes for topic-quotes-post and so on). In that case we won't need _handleMessages_ and _defineHandlerByTopic_ at all.
here's where we an do it: https://github.com/mojaloop/quoting-service/pull/321/files#diff-db88d0fcec394bf32e40eff7c738649415d22ee74c4c56577c8814c0c599ff6dR21

Lt's have a call and discuss if we need to do such optimization